### PR TITLE
minishift 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by gastaldi**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by maxim-belkin** [comment]
@gastaldi if this formula is added to Homebrew/core, we'll port it to Linuxbrew/core (with proper modifications).

**Comment by gastaldi** [comment]
@maxim-belkin Excellent. 

I have provided a PR in https://github.com/Linuxbrew/homebrew-core/pull/3011 too with this changes specific to Linux. Feel free to close it if that is not needed atm

**Comment by gastaldi** [comment]
It seems that it already exists as a Cask file. IMHO I think minishift should be available as a Formula, since it does not have any GUI to be a Cask (and also Casks are macOS only). 
The perfect approach would be to delete the Cask file, but I am not sure if that is allowed?

**Comment by commitay** [comment]
> The perfect approach would be to delete the Cask file

https://github.com/caskroom/homebrew-cask/pull/35760

**Comment by gastaldi** [comment]
Thanks @commitay

**Comment by commitay** [comment]
> IMHO I think minishift should be available as a Formula, since it does not have any GUI to be a Cask (and also Casks are macOS only).

@gastaldi From a previous PR to `homebrew-core` for `minishift` https://github.com/Homebrew/homebrew-core/issues/4531,  it was added to cask because it is distributed as a pre-built binary.

**Comment by gastaldi** [comment]
@commitay from the https://github.com/Homebrew/homebrew-core/pull/2417 comments, it seems that it was because in previous versions it depended on minikube and docker, which I believe is not the case anymore

**Review by ilovezfs — CHANGES_REQUESTED** [review]
The formula would need to build the software from source. Is that now possible without needing a Docker container?

**Comment by gastaldi** [comment]
@ilovezfs Why is building from source a requirement? The formula works as it is now.

For example, the jboss-forge formula provides a distribution ZIP file with the binary JARs and doesn't build from sources.

**Comment by ilovezfs** [comment]
@gastaldi this is a matter of policy that applies to everything in Homebrew.

**Comment by ilovezfs** [comment]
>For example, the jboss-forge formula provides a distribution ZIP file with the binary JARs and doesn't build from sources.

Yes, for cross-platform binaries such as Java jar files and Mono exe files there is a specific exception.

**Comment by ilovezfs** [comment]
@gastaldi see the documentation here: https://github.com/Homebrew/brew/blob/c541b3867ac7612cdfa3c7b8626367bea3e8f3ec/docs/Acceptable-Formulae.md#we-dont-like-binary-formulae

**Comment by gastaldi** [comment]
Thanks for the explanation. I'll try to adjust it to follow https://docs.openshift.org/latest/minishift/developing/developing.html#build-minishift-binary then, but my Ruby knowledge is somewhat limited, so I appreciate any help if possible :)

**Comment by ilovezfs** [comment]
@gastaldi it will need a better test, but this should get you started:
```
require "language/go"

class Minishift < Formula
  desc "Runs a single-node OpenShift cluster inside a VM"
  homepage "https://minishift.io/"
  url "https://github.com/minishift/minishift/archive/v1.1.0.tar.gz"
  sha256 "67cca4ec4767f2609eadb05f2fe3322592a070319caad2e3c477acbf68e0bff0"

  depends_on "go" => :build
  depends_on "glide" => :build

  go_resource "github.com/jteeuwen/go-bindata" do
    url "https://github.com/jteeuwen/go-bindata.git",
        :revision => "a0ff2567cfb70903282db057e799fd826784d41d"
  end

  def install
    ENV["GOPATH"] = buildpath
    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"

    (buildpath/"src/github.com/minishift/minishift").install buildpath.children
    ENV.prepend_create_path "PATH", buildpath/"bin"
    Language::Go.stage_deps resources, buildpath/"src"

    cd "src/github.com/jteeuwen/go-bindata/go-bindata" do
      system "go", "install"
    end

    cd "src/github.com/minishift/minishift" do
      system "make", "V=1"
      prefix.install_metafiles
    end

    bin.install "bin/minishift"
  end

  test do
    assert_equal "minishift v#{version}\n", shell_output("#{bin}/minishift version")
  end
end
```

**Comment by gastaldi** [comment]
So I came with this Formula: 
```
require "language/go"

class Minishift < Formula
  version "1.1.0"
  sha256 "67cca4ec4767f2609eadb05f2fe3322592a070319caad2e3c477acbf68e0bff0"

  desc "Runs a single-node OpenShift cluster inside a VM"
  homepage "https://minishift.io/"
  url "https://github.com/minishift/minishift/archive/v#{version}.tar.gz"

  depends_on "go" => :build
  depends_on "glide" => :build

  go_resource "github.com/jteeuwen/go-bindata" do
    url "https://github.com/jteeuwen/go-bindata.git",
        :revision => "a0ff2567cfb70903282db057e799fd826784d41d"
  end

  def install
    ENV["GOPATH"] = buildpath
    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"

    (buildpath/"src/github.com/minishift/minishift").install buildpath.children
    ENV.prepend_create_path "PATH", buildpath/"bin"
    Language::Go.stage_deps resources, buildpath/"src"

    cd "src/github.com/jteeuwen/go-bindata/go-bindata" do
      system "go", "install"
    end

    cd "src/github.com/minishift/minishift" do
      system "make"
      prefix.install_metafiles
    end

    bin.install "bin/minishift"
  end

  test do
    assert_equal "minishift v#{version}\n", shell_output("#{bin}/minishift version")
  end
end
```

But unfortunately I get a build error that I don't understand what's going on (The minishift build works fine if I follow the steps defined in https://docs.openshift.org/latest/minishift/developing/developing.html#build-minishift-binary): 

```
➜  Formula git:(minishift) ✗ brew install --build-from-source minishift.rb             
==> Downloading https://github.com/minishift/minishift/archive/v1.1.0.tar.gz
Already downloaded: /home/ggastald/.cache/Homebrew/minishift-1.1.0.tar.gz
==> Cloning https://github.com/jteeuwen/go-bindata.git
Updating /home/ggastald/.cache/Homebrew/minishift--github.com-jteeuwen-go-bindata--git
==> Checking out revision a0ff2567cfb70903282db057e799fd826784d41d
==> go install
==> make
Last 15 lines from /home/ggastald/.cache/Homebrew/Logs/minishift/02.make:
/usr/share/rubygems/rubygems/path_support.rb:83:in `default_path': undefined method `+' for nil:NilClass (NoMethodError)
	from /usr/share/rubygems/rubygems/path_support.rb:75:in `path='
	from /usr/share/rubygems/rubygems/path_support.rb:35:in `initialize'
	from /usr/share/rubygems/rubygems.rb:329:in `new'
	from /usr/share/rubygems/rubygems.rb:329:in `paths'
	from /usr/share/rubygems/rubygems.rb:352:in `path'
	from /usr/share/rubygems/rubygems/specification.rb:979:in `dirs'
	from /usr/share/rubygems/rubygems/specification.rb:831:in `stubs'
	from /usr/share/rubygems/rubygems/specification.rb:1036:in `find_by_path'
	from /usr/share/rubygems/rubygems.rb:189:in `try_activate'
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:125:in `rescue in require'
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:40:in `require'
	from <internal:gem_prelude>:6:in `<internal:gem_prelude>'
Makefile:56: recipe for target '/tmp/minishift-20170626-24138-8k0ikd/minishift-1.1.0/bin/minishift' failed
make: *** [/tmp/minishift-20170626-24138-8k0ikd/minishift-1.1.0/bin/minishift] Error 2

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
minishift 1.1.0 https://github.com/Linuxbrew/homebrew-core/pull/3011

```

**Comment by rwhogg** [comment]
> Do not report this issue to Homebrew/brew or Homebrew/core!

I looked at the three Jenkins builds, and from what I saw they all built correctly and only failed for audit reasons. So this may just be an error specific to Linuxbrew, in which case we* can deal with it there at the time this formula is merged into Linuxbrew.

* "We" in this context meaning you and the Linuxbrew team

**Comment by gastaldi** [comment]
@maxim-belkin any idea why that fails in Linuxbrew?

**Comment by gastaldi** [comment]
@rwhogg Thanks, I fixed the audit issues in a new commit and I hope it's OK now.

File: Formula/minishift.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require "language/go"
+
+class Minishift < Formula
+  desc "Runs a single-node OpenShift cluster inside a VM"
+  homepage "https://minishift.io/"
+  url "https://github.com/minishift/minishift/archive/v1.1.0.tar.gz"
+  version "1.1.0"
```
Comment (ilovezfs) [inline Formula/minishift.rb]: This line should be removed.

File: Formula/minishift.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    cd "src/github.com/minishift/minishift" do
+      system "make"
+      prefix.install_metafiles
+    end
+
+    bin.install "bin/minishift"
+  end
+
+  test do
+    assert_equal "minishift v#{version}\n", shell_output("#{bin}/minishift version")
```
Comment (ilovezfs) [inline Formula/minishift.rb]: We'll need a test that exercises and verifies the functionality of the software once installed.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
We'll need to sort out the name collision with the Cask. Are they the same software?

File: Formula/minishift.rb (hunk 2 above)
Comment (gastaldi) [inline Formula/minishift.rb]: Yes, they are the same software. There is even a PR removing it once this PR is merged: https://github.com/caskroom/homebrew-cask/pull/35760

File: Formula/minishift.rb (hunk 1 above)
Comment (gastaldi) [inline Formula/minishift.rb]: The version value is used in the tests. Is it still set even if I don't specify it?

**Comment by gastaldi** [comment]
Yes, they are the same software. There is even a PR removing it once this PR is merged: caskroom/homebrew-cask#35760

**Comment by ilovezfs** [comment]
>Yes, they are the same software. There is even a PR removing it once this PR is merged: caskroom/homebrew-cask#35760

Cool.

File: Formula/minishift.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/minishift.rb]: Yeah. See the audit error about its being redundant since it was already determined to be the same value from the URL.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Please refresh this with 1.4.1

**Comment by gastaldi** [comment]
Minishift now has its own update mechanism through `minishift update`, therefore I think it's fine to close this PR. Thanks for the help!

**Comment by gbraad** [comment]
@gastaldi This PR is abandoned, and Cask will stay in place?
If so, we will try to automate the process of pushing a changed cask when we cut a new release.

See: https://github.com/minishift/minishift/issues/1012

**Comment by gastaldi** [comment]
@gbraad yes, I think that's fine. I was using `minishift update` instead

**Pull request closed without merging**